lib/improver/tests/test_weighted_blend_BasicWeightedAverage.py

```diff
@@ -0,0 +1,165 @@
+        data[1][:][:] = 2.0
+        cube = Cube(data, standard_name="precipitation_amount",
+                    units="kg m^-2 s^-1")
+        cube.add_dim_coord(DimCoord(np.linspace(-45.0, 45.0, 2), 'latitude',
+                                    units='degrees'), 1)
+        cube.add_dim_coord(DimCoord(np.linspace(120, 180, 2), 'longitude',
+                                    units='degrees'), 2)
+        time_origin = "hours since 1970-01-01 00:00:00"
+        calendar = "gregorian"
+        tunit = Unit(time_origin, calendar)
+        cube.add_aux_coord(AuxCoord([402192.5, 402193.5],
+                                    "time", units=tunit), 0)
+        self.cube = cube
+        new_scalar_coord = iris.coords.AuxCoord(1,
+                                                long_name='dummy_scalar_coord',
+                                                units='no_unit')
+        cube_with_scalar = cube.copy()
+        cube_with_scalar.add_aux_coord(new_scalar_coord)
+        self.cube_with_scalar = cube_with_scalar
+
+    def test_basic(self):
+        """Test that the plugin returns an iris.cube.Cube."""
+        coord = "time"
+        plugin = BasicWeightedAverage(coord)
+        result = plugin.process(self.cube)
+        self.assertIsInstance(result, Cube)
+
+    def test_fails_coord_not_in_cube(self):
+        """Test it Raises a Value Error if coord not in the cube"""
+        coord = "notset"
+        plugin = BasicWeightedAverage(coord)
+        msg = ('The coord for this plugin must be ' +
+               'an existing coordinate in the input cube')
+        with self.assertRaisesRegexp(ValueError, msg):
+            plugin.process(self.cube)
+
+    def test_fails_input_not_a_cube(self):
+        """Test it Raises a Value Error if not supplied with a cube"""
+        coord = "time"
+        plugin = BasicWeightedAverage(coord)
+        notacube = 0.0
+        msg = ('The first argument must be an instance of ' +
+               'iris.cube.Cube')
+        with self.assertRaisesRegexp(ValueError, msg):
+            plugin.process(notacube)
+
+    def test_fails_weights_shape(self):
+        """Test it Raises a Value Error if weights shape does not match
+           coord shape"""
+        coord = "time"
+        plugin = BasicWeightedAverage(coord)
+        weights = [0.1, 0.2, 0.7]
+        msg = ('The weights array must match the shape ' +
+               'of the coordinate in the input cube')
+        with self.assertRaisesRegexp(ValueError, msg):
+            plugin.process(self.cube, weights)
+
+    def test_coord_adjust_set(self):
+        """Test it works with coord adjust set"""
+        coord = "time"
+        coord_adjust = lambda pnts: pnts[len(pnts)-1]
+        plugin = BasicWeightedAverage(coord, coord_adjust)
+        result = plugin.process(self.cube)
+        self.assertAlmostEquals(result.coord(coord).points, [402193.5])
+
+    def test_scalar_coord(self):
+        """Test it works on scalar coord"""
```

Comment:
It would be worth slightly expanding this docstring to capture that you're checking that a warning is being raised, if the dimension that you want to collapse on, is a scalar coordinate.